python/Ganga/test/GPI/FileTests/TestLocalFileClient.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+from Ganga.testlib.GangaUnitTest import GangaUnitTest
+from Ganga.testlib.file_utils import generate_unique_temp_file
+
+import datetime
+import time
+import os
+import shutil
+import copy
+
+class TestLocalFileClient(GangaUnitTest):
+    """test for sjid in filename names explain each test"""
+
+    _managed_files = []
+
+    # Num of sj in tests
+    sj_len = 3
+
+    # This sets up a LocalFileConfiguration which works by placing a file on local storage somewhere we can test using standard tools
+    LocalFileConfig = {'fileExtensions': [''],
+                       'uploadOptions': {},
+                       'backendPostprocess': {'LSF': 'client', 'LCG': 'client', 'ARC': 'client', 'Dirac': 'client',
+                                              'PBS': 'client', 'Interactive': 'client', 'Local': 'client', 'CREAM': 'client'}}
+
+    def setUp(self):
+        """
+        Configure the LocalFile for the test
+        """
+        extra_opts=[('PollThread', 'autostart', 'False'),
+                    ('Local', 'remove_workdir', 'False'),
+                    ('TestingFramework', 'AutoCleanup', 'False'),
+                    ('Output', 'LocalFile', self.LocalFileConfig),
+                    ('Output', 'FailJobIfNoOutputMatched', 'True')]
+        super(TestLocalFileClient, self).setUp(extra_opts=extra_opts)
+
+    @staticmethod
+    def cleanUp():
+        """ Cleanup the current temp objects """
+
+        from Ganga.GPI import jobs
+        for j in jobs:
+            shutil.rmtree(j.backend.workdir, ignore_errors=True)
+            j.remove()
+
+        for file_ in TestLocalFileClient._managed_files:
+            os.unlink(file_)
+        TestLocalFileClient._managed_files = []
+
+    def test_a_testClientSideSubmit(self):
+        """Test the client side code whilst stil using the Local backend"""
+
+        from Ganga.GPI import LocalFile, Job, ArgSplitter
+
+        TestLocalFileClient.cleanUp()
+
+        _ext = '.root'
+        file_1 = generate_unique_temp_file(_ext)
+        file_2 = generate_unique_temp_file(_ext)
+        TestLocalFileClient._managed_files.append(file_1)
+        TestLocalFileClient._managed_files.append(file_2)
+
+        j = Job()
+        j.inputfiles = [LocalFile(file_1), LocalFile(file_2)]
+        j.splitter = ArgSplitter(args = [[_] for _ in range(0, TestLocalFileClient.sj_len) ])
+        j.outputfiles = [LocalFile(namePattern='*'+_ext)]
+        j.submit()
+
+    def test_b_testClientSideComplete(self):
```

Comment:
Because debugging a problem with submitting a job and completing a job calls such different code that it's easier to navigate when they're in separate tests why they fail. By the same logic why not just make the whole file 1 test surely?